examples/tools/generate.ipynb

```diff
@@ -0,0 +1,502 @@
+    "resonance $\\omega \\to \\gamma\\pi^0$. This resonance has such a narrow peak\n",
+    "that we have to widen it to be visible and, even then, using an evenly\n",
+    "distributed phase space would not be feasible for a hit & miss strategy:\n",
+    "we want to have a higher density of phase-space points near the resonance.\n",
+    "\n",
+    "In case of a narrow resonance, **importance sampling** is beneficial. Instead\n",
+    "of creating an unweighted uniform phase space sample, we generate a weighted\n",
+    "sample with more points in the peak regions."
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "We first again create an amplitude model, this time including a 'widened'\n",
+    ":math:`\\omega` resonance (see also :doc:`particles`):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pycompwa.expertsystem.ui.system_control import StateTransitionManager\n",
+    "from pycompwa.expertsystem.ui.system_control import InteractionTypes\n",
+    "from pycompwa.expertsystem.amplitude.helicitydecay import \\\n",
+    "    HelicityAmplitudeGeneratorXML\n",
+    "from pycompwa.expertsystem.state.particle import particle_list\n",
+    "\n",
+    "initial_state = [(\"J/psi\", [-1, 1])]\n",
+    "final_state = [(\"gamma\"), (\"pi0\"), (\"pi0\")]\n",
+    "tbd_manager = StateTransitionManager(initial_state, final_state,\n",
+    "                                     formalism_type='helicity',\n",
+    "                                     topology_building='isobar')\n",
+    "\n",
+    "# Widen omega resonance\n",
+    "omega = particle_list[\"omega(782)\"]\n",
+    "parameters = omega['DecayInfo']['Parameter']\n",
+    "for par in parameters:\n",
+    "    if par['@Type'] == 'Width':\n",
+    "        par['Value'] = 0.001\n",
+    "\n",
+    "tbd_manager.set_allowed_interaction_types(\n",
+    "    [InteractionTypes.Strong, InteractionTypes.EM])\n",
+    "tbd_manager.allowed_intermediate_particles = ['f2(1270)', 'omega']\n",
+    "graph_interaction_settings_groups = tbd_manager.prepare_graphs()\n",
+    "solutions, _ = tbd_manager.find_solutions(graph_interaction_settings_groups)\n",
+    "\n",
+    "print(\"found \" + str(len(solutions)) + \" solutions!\")\n",
+    "\n",
+    "xml_generator = HelicityAmplitudeGeneratorXML()\n",
+    "xml_generator.generate(solutions)\n",
+    "xml_generator.write_to_file('model.xml')"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "Now we generate a phase space sample which is importance sampled by the\n",
+    "intensity. This can be done with the\n",
+    ":func:`.generate_importance_sampled_phsp` function. Note that you still\n",
+    "need a 'normal' phase-space sample for the :class:`.IntensityBuilderXML`."
```

Comment:
Hmm, this is a bit tricky. In practice you don't need the phase space sample, and you should not use one. Unfortunately the expert system automatically uses the `NormalizedIntensity` tag for the outermost intensity, which requires this. To normalized this correctly a importance sample would be needed... which is what we want to generate....
Luckily since this is data generation and the parameters of the model stay fixed this normalization is irrelevant and just a constant scaling factor. So best remove the `NormalizedIntensity` tag or use some "dummy" phase space sample with 1 event or so. This is bad design btw